tuf/api/metadata.py

```diff
@@ -728,6 +728,135 @@ def update(
             self.meta[metadata_fn]["hashes"] = hashes
 
 
+class DelegationRole(Role):
+    """A container with information about particular delegated role.
+
+    Attributes:
+        name: A string giving the name of the delegated role.
+        keyids: A set of strings each of which represents a given key.
+        threshold: An integer representing the required number of keys for that
+            particular role.
+        terminating: A boolean indicating whether subsequent delegations
+            should be considered.
+        paths: An optional list of strings, where each string describes
+            a path that the role is trusted to provide.
+        path_hash_prefixes: An optional list of HEX_DIGESTs used to succinctly
+            describe a set of target paths. Only one of the attributes "paths"
+            and "path_hash_prefixes" is allowed to be set.
+        unrecognized_fields: Dictionary of all unrecognized fields.
+
+    """
+
+    def __init__(
+        self,
+        name: str,
+        keyids: list,
+        threshold: int,
+        terminating: bool,
+        paths: Optional[list] = None,
+        path_hash_prefixes: Optional[list] = None,
+        unrecognized_fields: Optional[Mapping[str, Any]] = None,
+    ) -> None:
+        super().__init__(keyids, threshold, unrecognized_fields)
+        self.name = name
+        self.terminating = terminating
+        if paths and path_hash_prefixes:
+            raise ValueError(
+                "Only one of the attributes 'paths' and"
+                "'path_hash_prefixes' can be set!"
+            )
+        if paths:
+            self.paths = paths
+        elif path_hash_prefixes:
+            self.path_hash_prefixes = path_hash_prefixes
```

Comment:
if-else leads to undefined object variables. Some variant of this is not tested.

Reply:
the instance attributes are left unset so I think `to_dict() `will fail in some of those cases (at least when you only have path_hash_prefixes)

Reply:
Just tested it and you are right. We don't need the if-else here.